### ISSUE TITLE
package.json has no "main" entry error and webiny module error when attempting to deploy local dev environment.

### ISSUE THREAD

**Issue by TommyJackson85** [webiny/webiny-js#984]
## This is:

- Bug

## Specifications

  - version:
  - OS:

## Expected Behavior
Deploy application following prerequisites and local setup guidlines.

## Actual Behavior
Fails running ```npx webiny deploy api --env=local``` in the end reporting an error.
Seems to be a webiny module error but ```@webiny``` exists in my node modules.
Also reporting that ```package.json``` has no "main" entry so it could be in the package.json file.

## Steps to Reproduce the Problem

1. Follow the guidelines  for setting up a [local development environment](https://github.com/webiny/webiny-js/blob/develop/docs/CONTRIBUTING.md).

## Detailed Description

Full bug report -
```
(node:11281) UnhandledPromiseRejectionWarning: Error: Cannot find module '/Users/thomasjackson/Desktop/Software projects/Open Source Projects/Webiny/webiny-js/node_modules/@webiny/plugins/index.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (internal/modules/cjs/loader.js:228:19)
    at Function.Module._findPath (internal/modules/cjs/loader.js:365:18)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:610:27)
    at Function.Module._load (internal/modules/cjs/loader.js:527:27)
    at Module.require (internal/modules/cjs/loader.js:681:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at Object.<anonymous> (/Users/thomasjackson/Desktop/Software projects/Open Source Projects/Webiny/webiny-js/packages/cli/context.js:6:30)
    at Module._compile (internal/modules/cjs/loader.js:774:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:785:10)
    at Module.load (internal/modules/cjs/loader.js:641:32)
(node:11281) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an asyncfunction without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:11281) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

## Possible Solution
Update webiny node package or is there something I am missing?
Didn't use npm to setup node modules, just yarn.

**Comment by TommyJackson85** [comment]
@Pavel910 Yes I am setting up a repository. I am following that guide and I am running ```npx webiny deploy api --env=local``` from the /sample-project file.
Is there anything else I could have missed ?

**Comment by Pavel910** [comment]
@TommyJackson85 when you setup the repo using `yarn setup-repo`, the packages are built and linked via `node_modules`. You can actually open your `<root>/node_modules/@webiny` and see if the repo packages are linked via symlinks:

![image](https://user-images.githubusercontent.com/3920893/84738976-16805900-afab-11ea-8412-7da8ee9b056c.png)

Note the small `symlink` icon. If those links are not there, do the following from the root of the repo:
```
yarn build
node scripts/linkPackages,js
```

You can learn more about the repo architecture in this document: https://github.com/webiny/webiny-js/blob/develop/docs/ARCHITECTURE_AND_CONCEPTS.md

**Comment by TommyJackson85** [comment]
Ok I ran ```yarn setup-repo``` (from webiny-js folder) again and I realised there is an error I didn't take notice of, which also appears from running ```yarn build``` (from webiny-js folder). The error is to do with ```webiny/commodo```:

```
@webiny/commodo: src/fields-float.ts(1,15): error TS2307: Cannot find module 'commodo-fields-float'.
@webiny/commodo: src/fields-int.ts(1,15): error TS2307: Cannot find module 'commodo-fields-int'.
@webiny/commodo: src/index.ts(8,15): error TS2307: Cannot find module 'commodo-fields-int'.
@webiny/commodo: src/index.ts(9,15): error TS2307: Cannot find module 'commodo-fields-float'.
@webiny/commodo: npm ERR! code ELIFECYCLE
@webiny/commodo: npm ERR! errno 2
@webiny/commodo: npm ERR! @webiny/commodo@4.0.2 postbuild: `cp package.json LICENSE README.md dist/ && tsc -p tsconfig.build.json`
@webiny/commodo: npm ERR! Exit status 2
@webiny/commodo: npm ERR!
@webiny/commodo: npm ERR! Failed at the @webiny/commodo@4.0.2 postbuild script.
@webiny/commodo: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
@webiny/commodo: npm WARN Local package.json exists, but node_modules missing, did you mean to install?
@webiny/commodo: npm ERR! A complete log of this run can be found in:
@webiny/commodo: npm ERR!     /Users/thomasjackson/.npm/_logs/2020-06-16T19_50_00_427Z-debug.log
lerna ERR! npm run build exited 2 in '@webiny/commodo'
lerna WARN complete Waiting for 8 child processes to exit. CTRL-C to exit immediately.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Anything I could of missed?

**Comment by Pavel910** [comment]
Commodo packages should be in your `node_modules`, they are not part of the repo. We install them as regular `npm` dependencies, so I have this in my `node_modules`:

![image](https://user-images.githubusercontent.com/3920893/84824089-b11a7f80-b01f-11ea-9e18-47c1af840999.png)

Double check that manually, to be sure you have them. It's a little bit weird, but hopefully we'll figure out what's wrong.

Btw. an important note: as of today, we no longer have the `develop` branch, we now only have `master` and that is our development branch with latest code. Just info to be aware of.

Let me know about those `commodo-*` packages. If you don't have them, remove `yarn.lock` in your repo and run `yarn` again. Watch for errors.

Cheers!

**Comment by TommyJackson85** [comment]
Here were my npm packages: 
<img width="344" alt="Screenshot 2020-06-16 at 21 46 02" src="https://user-images.githubusercontent.com/22279028/84828977-df4b8f80-b01e-11ea-8c21-3aa98a922633.png">
<img width="303" alt="Screenshot 2020-06-16 at 21 42 48" src="https://user-images.githubusercontent.com/22279028/84828994-e7a3ca80-b01e-11ea-8e9d-e60d8801940a.png">
So you were right! I was missing those packages. so I removed yarn.lock, then ran ```yarn``` then ```yarn setup-repo```
then from sample-project I ran ```npx webiny deploy api --env=local```
And the initial error is gone, :) its attempting to deploy.
If I come across another issue with deployment i'll post it as a separate issue, but keep this issue open for a day incase the error reappears maybe.

Thanks @pavel910 !
I don't know how I caused this but maybe it's worth writing this up in the Contributions documentation if others are coming across it. Maybe highlight that if the error occurs, delete yarn.lock and restart.
